src/drivers/osd/msp_osd/uorb_to_msp.cpp

```diff
@@ -314,9 +314,9 @@ msp_raw_gps_t construct_RAW_GPS(const sensor_gps_s &vehicle_gps_position,
 	msp_raw_gps_t raw_gps {0};
 
 	if (vehicle_gps_position.fix_type >= 2) {
-		raw_gps.lat = vehicle_gps_position.lat;
-		raw_gps.lon = vehicle_gps_position.lon;
-		raw_gps.alt =  vehicle_gps_position.alt / 10;
+		raw_gps.lat = static_cast<int32_t>(vehicle_gps_position.latitude_deg * 1e7);
+		raw_gps.lon = static_cast<int32_t>(vehicle_gps_position.longitude_deg * 1e7);
+		raw_gps.alt = static_cast<int16_t>(vehicle_gps_position.altitude_msl_m);
```

Comment:
```suggestion
		raw_gps.alt = static_cast<int16_t>(vehicle_gps_position.altitude_msl_m*100);
```

Reply:
I am working on the rest of the changes (accepting all) - but this one needs attention. I think original division by 10 (mm->cm conversion) was conflicting with the units specifies in the _msp_raw_gps_t_  message (meters):

src/drivers/osd/msp_osd/uorb_to_msp.cpp : 313
```
	// initialize result
	msp_raw_gps_t raw_gps {0};

	if (vehicle_gps_position.fix_type >= 2) {
		raw_gps.lat = static_cast<int32_t>(vehicle_gps_position.latitude_deg * 1e7);
		raw_gps.lon = static_cast<int32_t>(vehicle_gps_position.longitude_deg * 1e7);
		raw_gps.alt = static_cast<int16_t>(vehicle_gps_position.altitude_msl_m);
```
src/drivers/osd/msp_osd/msp_defines.h : 383
```
// MSP_RAW_GPS reply
struct msp_raw_gps_t {
	uint8_t  fixType;       // MSP_GPS_NO_FIX, MSP_GPS_FIX_2D, MSP_GPS_FIX_3D
	uint8_t  numSat;
	int32_t  lat;           // 1 / 10000000 deg
	int32_t  lon;           // 1 / 10000000 deg
	int16_t  alt;           // meters
	int16_t  groundSpeed;   // cm/s
	int16_t  groundCourse;  // unit: degree x 100, centidegrees
	uint16_t hdop;
} __attribute__((packed));
```
What is your decision on this? is it meters or centimeters that we should pass to msp?

Reply:
Hmm it looks like the specification changed: https://github.com/betaflight/betaflight/blob/master/src/main/msp/msp.c#L1522.
Probably best to use the new one then, and update the comment in the struct.

Reply:
Understood, will update comment in _msp_defines.h_ ("meters"->"centimeters since MSP API 1.39, meters before") and accept the "/100" in _uorb_to_msp.cpp_